### DISCUSSION TITLE
How to add custom ` to Enums

### DISCUSSION THREAD

**Discussion by yordis** [fastapi/fastapi#10772] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from enum import Enum

class UserIntention(Enum):
    OPEN = "OPEN"
    CLOSE = "CLOSE"
    NONE = "NONE"

    model_config = {"json_schema_extra": {"x-schema-id": "UserIntention"}}
```

### Description

Hey folks, first, I never done Python before, so my understand is quite limited, I tried by best to research the entire internet before I asked for help.

I am having an issue where `all` the schemas under `#components.schemas` in the OpenAPI spec MUST have a `x-schema-id` field.

I got the normal `BaseModel` to work, but I can not find the way I can add the `json_schema_extra` to the Enums.

### Operating System

macOS

### FastAPI Version

0.104.1

### Pydantic Version

2.5.2

### Python Version

Python 3.10.13

**Comment by Ventura94** [comment]
Can you give more detail with examples to better understand what you need?
